conversion.go

```diff
@@ -465,6 +465,46 @@ func bigEndianUint56(b []byte) uint64 {
 		uint64(b[2])<<32 | uint64(b[1])<<40 | uint64(b[0])<<48
 }
 
+// MarshalSSZTo implements the fastssz.Marshaler interface and serializes the
+// integer into an already pre-allocated buffer.
+func (z *Int) MarshalSSZTo(dst []byte) ([]byte, error) {
+	if len(dst) != 32 {
+		return nil, fmt.Errorf("%w: have %d, want %d bytes", ErrBadBufferLength, len(dst), 32)
+	}
```

Comment:
```suggestion
	if len(dst) < 32 {
		return nil, fmt.Errorf("%w: have %d, want >= %d bytes", ErrBadBufferLength, len(dst), 32)
	}
```

Reply:
That would accept and ignore junk at the end. I'd guess we want to enforce struct parsing.